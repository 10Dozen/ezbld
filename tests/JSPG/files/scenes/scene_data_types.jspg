# SceneParamsAsStrings
*goto: ValueAsString
*pre_exec: ValueAsString
*post_exec: ValueAsString

# SceneParamsAsVars
*goto: `Value.As.Variable`
*pre_exec: `Value.As.Variable`
*post_exec: `Value.As.Variable`

# SceneParamsAsCode
*goto: { return Value.As.Variable }
*pre_exec: { return Value.As.Variable }
*post_exec: { return Value.As.Variable }

# SceneParamsAsWrappedCode
**goto:
    return Value.As.Variable
**pre_exec:
    return Value.As.Variable
**post_exec:
    return Value.As.Variable
*
