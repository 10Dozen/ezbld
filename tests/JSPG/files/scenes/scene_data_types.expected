[0]
cls = JSPGScene
name = SceneParamsAsStrings | TEXT
goto = ValueAsString | TEXT
pre_exec = ValueAsString | TEXT
post_exec = ValueAsString | TEXT

[1]
cls = JSPGScene
name = SceneParamsAsVars | TEXT
goto = Value.As.Variable | VARIABLE
pre_exec = Value.As.Variable | VARIABLE
post_exec = Value.As.Variable | VARIABLE


[2]
cls = JSPGScene
name = SceneParamsAsCode | TEXT
goto = ()=>{ return Value.As.Variable } | FUNCTION
pre_exec = ()=>{ return Value.As.Variable } | FUNCTION
post_exec = ()=>{ return Value.As.Variable } | FUNCTION


[3]
cls = JSPGScene
name = SceneParamsAsWrappedCode | TEXT
goto =
    ()=>{
    \treturn Value.As.Variable
    } | MULTILINE_FUNCTION
pre_exec =
    ()=>{
    \treturn Value.As.Variable
    } | MULTILINE_FUNCTION
post_exec =
    ()=>{
    \treturn Value.As.Variable
    } | MULTILINE_FUNCTION
